components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/execution/DependencyInstallerImpl.java

```diff
@@ -0,0 +1,265 @@
+import org.wso2.carbon.siddhi.extensions.installer.core.exceptions.ExtensionsInstallerException;
+import org.wso2.carbon.siddhi.extensions.installer.core.models.enums.ExtensionInstallationStatus;
+import org.wso2.carbon.siddhi.extensions.installer.core.models.enums.ExtensionUnInstallationStatus;
+import org.wso2.carbon.siddhi.extensions.installer.core.util.ResponseEntityCreator;
+import org.wso2.carbon.siddhi.extensions.installer.core.util.ExtensionsInstallerUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Handles installation and un-installation of extensions.
+ */
+public class DependencyInstallerImpl implements DependencyInstaller {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(DependencyInstallerImpl.class);
+
+	/**
+	 * Extension configurations, denoted by extension Id.
+	 */
+	private final Map<String, ExtensionConfig> extensionConfigs;
+
+	public DependencyInstallerImpl(Map<String, ExtensionConfig> extensionConfigs) {
+		this.extensionConfigs = extensionConfigs;
+	}
+
+	@Override
+	public Map<String, Object> installDependenciesFor(String extensionId) throws ExtensionsInstallerException {
+		List<DependencyConfig> completedDependencies = new ArrayList<>();
+		List<DependencyConfig> failedDependencies = new ArrayList<>();
+		List<DependencyConfig> manuallyInstallableDependencies = new ArrayList<>();
+
+		ExtensionConfig extension = ExtensionsInstallerUtils.findExtension(extensionId, extensionConfigs);
+		List<DependencyConfig> dependencies = extension.getDependencies();
+		if (dependencies != null) {
+			for (DependencyConfig dependency : dependencies) {
+				if (dependency.isAutoDownloadable()) {
+					try {
+						installUsagesFor(dependency);
+						completedDependencies.add(dependency);
+					} catch (ExtensionsInstallerException e) {
+						failedDependencies.add(dependency);
+						LOGGER.error(String.format("Failed to install dependency: %s for extension: %s.",
+								dependency.getRepresentableName(), extensionId), e);
+					}
+				} else {
+					manuallyInstallableDependencies.add(dependency);
+				}
+			}
+			ExtensionInstallationStatus status = ExtensionsInstallerUtils.getInstallationStatus(
+					completedDependencies.size(), extension.getDependencies().size());
+			return ResponseEntityCreator.createExtensionInstallationResponse(
+					status, completedDependencies, failedDependencies, manuallyInstallableDependencies);
+		}
+		throw new ExtensionsInstallerException(String.format(
+				"No dependencies were specified in extension: %s", extensionId));
+	}
+
+	private void installUsagesFor(DependencyConfig dependency) throws ExtensionsInstallerException {
```

Comment:
Need change the method name to something which give more context

Reply:
Discussed offline, hence resolving.